ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -169,32 +279,598 @@ module Json = struct
     | _ ->
         [("event", `Null); ("session", `Null)]
 
+  let of_result obj msg =
+    match msg.msg_result with
+    | None ->
+        `Null
+    | Some (t, _d) ->
+        if obj.name = "event" && String.lowercase_ascii msg.msg_name = "from"
+        then
+          `O
+            [
+              ("type", `String "EventBatch")
+            ; ("func_partial_type", `String "EventBatch")
+            ]
+        else
+          let t', _ = string_of_ty_with_enums t ~action:Deserialize () in
+          `O
+            [
+              ("type", `String t')
+            ; ("func_partial_type", `String (get_fp_type t))
+            ]
+
+  let group_params params =
+    List.fold_left
+      (fun groups (param_release, info) ->
+        match groups with
+        | [] ->
+            [(param_release, [info])]
+        | group :: tl -> (
+          match group with
+          | pr, infos when pr = param_release ->
+              (pr, info :: infos) :: tl
+          | _ ->
+              (param_release, [info]) :: groups
+        )
+      )
+      [] params
+
+  let scan_left func init list =
+    let rec aux func init list result =
+      match list with
+      | [] ->
+          result
+      | x :: xs ->
+          let init = func init x in
+          aux func init xs (init :: result)
+    in
+    List.rev @@ aux func init list []
+
+  let combine_groups groups =
+    let scans =
+      scan_left
+        (fun (len, group) (_, infos) -> (len + List.length infos, group @ infos))
+        (0, []) groups
+    in
+    match scans with [] -> [] | [x] -> [x] | _ :: xs -> xs
+
+  let rec get_last = function
+    | [] ->
+        failwith "empty list"
```

Comment:
Include `__FUNCTION__` or `__LOC__` in the error to make it simpler to find. A better name would be simply `last` because every function could be prefixed with `get`.

Reply:
These related code has been split to the following PR which will raised after this one got merged.